assembly/src/library/mod.rs

```diff
@@ -15,6 +21,150 @@ pub use self::version::{Version, VersionError};
 #[cfg(test)]
 mod tests;
 
+// COMPILED LIBRARY
+// ===============================================================================================
+
+/// Represents a library where all modules modules were compiled into a [`MastForest`].
+pub struct CompiledLibrary {
```

Comment:
Only thing left would be to add metadata to the library, but I'm unclear on what that struct should look like